### PR TITLE
Update status bar dinamically 

### PR DESCRIPTION
**Pull request by asankov**
@atodorov I left two TODOs in the code. I would like to hear your opinion on them:

do we have this ID loaded somewhere, where I can use it and I am just not seeing it?
```javascript
 // TODO: better way to do this
 const testRunId = window.location.pathname.split('/')[2]
```
IMO this data structure is good enough for the purpose I am using it, but if you can think of a more elegant solution I will apply it
```javascript
// TODO: better solution
executionStatuses.forEach(s => statusCount[s.name] = { count: 0, id: s.id })
```

### CODE REVIEW

File: tcms/static/js/testrun_actions.js [hunk 1]
```diff
@@ -683,6 +654,59 @@ jQ(document).ready(function(){
...
+    const completePercent = completeCount / allCount * 100;
+    const failurePercent = failCount / completeCount * 100;
+    $(".progress-bar > .percent > .complete-percent").text(`${completePercent}%`)
+    $(".progress-bar > .progress-inner").css("width", `${completePercent}%`)
+    $(".progress-bar > .progress-inner > .progress-failed").css("width", `${failurePercent}%`)
+    $(".js-caserun-total").text(allCount)
+
+    $(".count-per-status-list").remove()
+    $(".count-per-status-container").prepend('<span class="count-per-status-list"></span>')
+    for (var status in statusCount) {
```
Comment (atodorov) [inline tcms/static/js/testrun_actions.js]: this looks like it can be a separate function. 
that will make the linters happier and is less prone to breakage when this page gets redesigned, which appears to be quite soon.

File: tcms/static/style/base.css [hunk 2]
```diff
@@ -128,6 +128,11 @@ hr.dotted{ border:dotted #ccc 0px;border-top-width:1px;}
 .commentdelete{float:right; display:block; background-position:0 -15px;padding-left:13px; margin:-20px 5px 0 0;cursor:pointer;}
 table.noborder,table.noborder td{border:none;background:transparent;}
 
+.link {
+    color: #004481 !important;
+    text-decoration: none;
+    cursor: pointer
+}
```
Comment (atodorov) [inline tcms/static/style/base.css]: The existing CSS classes seem to be doing the job of showing the pointer just right. Why do we need another CSS style  here ?

File: tcms/static/js/testrun_actions.js [hunk 3]
```diff
@@ -222,7 +190,7 @@ Nitrate.TestRuns.Details.on_load = function() {
   jQ('.js-caserun-total').bind('click', function() {
     showCaseRunsWithSelectedStatus(jQ('#id_filter')[0], '');
   });
-  jQ('.js-status-subtotal').bind('click', function() {
+  jQ('.status-link-button').bind('click', function() {
```
[the remark this answers is not in the document]
Comment (asankov) [reply tcms/static/js/testrun_actions.js]: weird. it does for me. can you try now with the latest changes?

File: tcms/static/style/base.css (hunk 2 above)
Comment (asankov) [reply tcms/static/style/base.css]: which classes do you mean exactly?

the existing version of this does not uses css classes, but `<a href="javascript:void(0)">` to do the job of styling

File: tcms/static/js/testrun_actions.js [hunk 4]
```diff
@@ -683,6 +654,59 @@ jQ(document).ready(function(){
...
+      const executionStatus = executionStatuses.find(s => s.id === testExecution.status_id)
+
+      if (executionStatus.weight > 0) {
+        successCount++
+      } else if (executionStatus.weight < 0) {
+        failCount++
+      }
+
+      statusCount[executionStatus.name].count++
+    })
```
[the remark this answers is not in the document]
Comment (asankov) [reply tcms/static/js/testrun_actions.js]: I'm not sure we could this this way. Maybe with some `.map` and `.filter`. I can try if you think this it will be better

File: tcms/static/js/testrun_actions.js [hunk 5]
```diff
@@ -683,6 +654,59 @@ jQ(document).ready(function(){
   });
 });
 
+function drawPercentBar() {
+
+  // TODO: better way to do this
+  const testRunId = window.location.pathname.split('/')[2]
```
[the remark this answers is not in the document]
Comment (asankov) [reply tcms/static/js/testrun_actions.js]: yes, thanks for the heads-up

File: tcms/static/js/testrun_actions.js [hunk 6]
```diff
@@ -683,6 +654,59 @@ jQ(document).ready(function(){
...
+
+  jsonRPC('TestExecution.filter', {'run_id': testRunId}, testExecutions => {
```
[the remark this answers is not in the document]
Comment (asankov) [reply tcms/static/js/testrun_actions.js]: but here we are not referring to the id of the `TestExecution`, but the ID of the run the execution is attached to.

File: tcms/static/style/base.css (hunk 2 above)
Comment (atodorov) [reply tcms/static/style/base.css]: The can't we use an `<a href="#">` tag instead of `<span>` when rendering this widget ? And no need of styling.

File: tcms/static/js/testrun_actions.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (atodorov) [reply tcms/static/js/testrun_actions.js]: You don't need map and filter, just regular calls to the API to tell you how many positive & negative statuses there are. Nevermind, works as it is so don't bother.

File: tcms/static/js/testrun_actions.js [hunk 7]
```diff
@@ -683,6 +655,62 @@ jQ(document).ready(function(){
...
+      } else if (executionStatus.weight < 0) {
+        negativeCount++
+      }
+
+      statusCount[executionStatus.name].count++
+    })
+  
+    const completeCount = positiveCount + negativeCount;
+    const completePercent = completeCount / allCount * 100;
+    const failurePercent = negativeCount / completeCount * 100;
```
Comment (atodorov) [inline tcms/static/js/testrun_actions.js]: Causes a regression wrt to how many decimal points are shown. Limit to only 2 decimal points at most.

File: tcms/templates/run/status_statistics.html [hunk 8]
```diff
@@ -2,32 +2,22 @@
...
-                    <label>{% trans "TOTAL" %}</label>
-			<span id="TOTAL">[</span><a href="javascript:void(0)" class="js-caserun-total">{{ status_stats.CaseRunsTotalCount }}</a><span>]</span>
+			<label>{% trans "TOTAL" %}</label>
+			[<a href="#" class="js-caserun-total"></a>]
 		</li>
 	</ul>
 	<div class="clear"></div>
 	<ul>
-            <li style="background:url({% static 'images/btn_bg_report.png' %}) no-repeat; height:25px; line-height:25px; width:57px; padding-left:30px; margin-right:15px;"><a href="{% url "run-report" test_run.run_id %}" title="report of test caserun">{% trans "Report" %}</a></li>
+		<li style="background:url({% static 'images/btn_bg_report.png' %}) no-repeat; height:25px; line-height:25px; width:57px; padding-left:30px; margin-right:15px;"><a href="{% url "run-report" test_run.run_id %}" title="report of test caserun">{% trans "Report" %}</a></li>
```
Comment (atodorov) [inline tcms/templates/run/status_statistics.html]: Unrelated changes causing conflicts.

**Comment by asankov** [comment]
> Need to rebase on top of #1355

Done.

> when the auto-next checkbox isn't selected status isn't updated

I am not able to reproduce now.
